kernel/src/hil/eic.rs

```diff
@@ -0,0 +1,55 @@
+//! Interface for external interrupt controller.
+
+/// Enum for selecting which edge to trigger interrupts on.
+#[derive(Debug)]
+pub enum InterruptMode {
+    RisingEdge,
+    FallingEdge,
+    HighLevel,
+    LowLevel,
+}
+
+/// Enum for enabling/disabling filter
+pub enum FilterMode {
+    FilterEnable,
+    FilterDisable,
+}
+
+/// Enum for selecting syn/asyn mode
+pub enum SynchronizationMode {
+    Synchronous,
+    Asynchronous,
+}
+
+/// Interface for EIC.
+pub trait ExternalInterruptController {
+    /// Enables EIC module
+    fn enable(&self);
```

Comment:
I think it would be best to remove the `enable()` and `disable()` functions, as I don't think we typically have them in Tock HILs. A user is unlikely to want to enable the controller but then not use it, or enable a specific interrupt but not enable the actual controller. The implementation of the HIL should automatically enable/disable the actual controller rather than requiring the user to manage that.